src/sentry/api/endpoints/client_state.py

```diff
@@ -50,12 +33,13 @@ class ClientStateEndpoint(OrganizationEndpoint):
     private = True
 
     def __init__(self, **options) -> None:
-        cluster_key = getattr(settings, "SENTRY_CLIENT_STATE_REDIS_CLUSTER", "default")
-        self.client = redis.redis_clusters.get(cluster_key)
+        self.client = get_redis_client()
         super().__init__(**options)
 
     def get(self, request: Request, organization, category) -> Response:
         key = get_client_state_key(organization.slug, category, request.user)
+        if not key:
```

Comment:
@Neo-Zhixing Looks like there is a lot of repeated code here. I suggest that you can handle this in `convert_args` where we can add the key as an argument to the request methods.